### DISCUSSION TITLE
Filter Is not working with the infinite scroll on Category Page Bigcommerce

### DISCUSSION THREAD

**Discussion by Mj6078** [community/community#145581] [Programming Help]
### Body

Did anyone know I have implemented the custom fields filter logic and the infinite scroll? The main problem is I have made logic where all unique custom fields will come as filters, and due to this only initial stage, it is 20 products. Still, then after it loads with the infinite scroll, the main issue is that the filter will only work with the first 20 products. However, then it will also show all custom field names in the filter, which is still not loaded, and due to this, when the user clicks on them, it shows up a blank screen with zero product, and this is the main issue in bigcommerce stencil cli. Can anyone help me out to solve this?

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Evinrrr** [comment]
The problem is that the custom fields filter logic doesn't work well with infinite scroll. Initially, only 20 products are loaded, and the filter only applies to them. After infinite scroll loads more, the filter still only affects the first 20. Also, clicking on unloaded custom fields shows a blank screen. To solve this, the filter logic needs to be adjusted to work with dynamically loaded products, and a check should be added to prevent showing blank screens when clicking on unloaded fields.

**Reply by mecodeatlas** [reply]
Hey, @Mj6078! Did @Evinrrr's response help you with your question? :)

**Reply by Mj6078** [reply]
> The problem is that the custom fields filter logic doesn't work well with infinite scroll. Initially, only 20 products are loaded, and the filter only applies to them. After infinite scroll loads more, the filter still only affects the first 20. Also, clicking on unloaded custom fields shows a blank screen. To solve this, the filter logic needs to be adjusted to work with dynamically loaded products, and a check should be added to prevent showing blank screens when clicking on unloaded fields.

Like i get it But can you give some idea how I can make logic on this statement?
